doc/specs/#2046 - Unified keybindings and commands, and synthesized action names.md

```diff
@@ -0,0 +1,589 @@
+While that spec is complete in it's own right, subsequent discussion revealed
+additional ways to improve the functionality and usability of the command
+palette. This document builds largely on the topics already introduced in the
+original spec, so readers should first familiarize themselves with that
+document.
+
+One point of note from the original document was that the original specification
+was entirely too verbose when defining both keybindings and commands for
+actions. Consider, for instance, a user that wants to bind the action "duplicate
+the current pane". In that spec, they need to add both a keybinding and a
+command:
+
+```json
+{
+    "keybindings": [
+        { "keys": [ "ctrl+alt+t" ], "command": { "action": "splitPane", "split":"auto", "splitMode": "duplicate" } },
+    ],
+    "commands": [
+        { "name": "Duplicate Pane", "action": { "action": "splitPane", "split":"auto", "splitMode": "duplicate" }, "icon": null },
+    ]
+}
+```
+
+These two entries are practically the same, except for two key differentiators:
+* the keybinding has a `keys` property, indicating which key chord activates the
+  action.
+* The command has a `name` property, indicating what name to display for the
+  command in the Command Palette.
+
+What if the user didn't have to duplicate this action? What if the user could
+just add this action once, in their `keybindings` or `commands`, and have it
+work both as a keybinding AND a command?
+
+
+## Solution Design
+
+This spec will outline two primary changes to keybindings and commands.
+1. Unify keybindings and commands, so both `keybindings` and `commands` can
+   specify either actions bound to keys, and/or actions bound to entries in the
+   Command Palette.
+2. Propose a mechanism by which actions do not _require_ a `name` to appear in
+   the Command Palette.
+
+
+### Proposal 1: Unify Keybindings and Commands
+
+As noted above, keybindings and commands have nearly the exact same syntax, save
+for a couple properties. To make things easier for the user, I'm proposing
+treating everything in _both_ the `keybindings` _and_ the `commands` arrays as
+**BOTH** a keybinding and a command.
+* Anything entry that has a `keys` value will be added to the keybindings.
+  Pressing that keybinding will activate the action defined in `command`.
+* Anything with a `name`<sup>[1]</sup> will be added as an entry (using that
+  name) to the Command Palette's Action Mode.
+* `keybindings` will be parsed first into keybindings and commands, then
+  `commands` will be layered on top.
+  - The decision to keep both `keybindings` and `commands` as top-level objects
+    in the json blob is certainly **_TODO_: UP FOR DISCUSSION**. Now that we're
+    adding things that might not be bound to keys to the array, `commands` seems
+    like a more appropriate name. However, we're definitely not going to be able
+    to rename `keybindings`.
+
+###### Caveats

```

Comment:
> there's 2 keybindings/commands with the same name

In this case, the second one will override the first. 
* If they're using the auto-generated names, this should only happen if they're the _same action_, so that's not interesting.
* If the user manually set the name of the second action to be the same as an auto-generated name, but with a different action, then that's _weird_, but hey, the user opted in to that behavior.
* If they're both custom names, then the second one will override the first.

> there's 2 keybindings/commands with the same keys

This is basically the same as before - a key can only be bound to one action, so whichever was the last keybinding/command to be bound to a given key will be the winner.

> what happens when...

Yes, "Copy Text" would stay in the command palette, without a key bound to it.